Feature/12875161 add save only at pictures folder mode

ericcleao opened the pull request.

ericcleao committed:
feature/12875161 Add save only mode front and back

- At front create the button and the change icon function
- Created the show and hide save only snackbar methods
- At back check the state and save at pictures depending of the actual save mode

Signed-off-by: Eric Cerqueira Leao <ericcleao@gmail.com>

ericcleao committed:
feature/12875161 Update save mode to use a constant class instead two static variables

- Create sva emode constant class
- Bring back the "just" method call to save on pictures folder method
- Create an snackbar to explain when user come back to stant save mode that those photo will be saved on the app
- Translate the strings that are been shown at the snackbars

Signed-off-by: Eric Cerqueira Leao <ericcleao@gmail.com>

ericcleao committed:
feature/12875161 Fix initial icon at change mode button

-

Signed-off-by: Eric Cerqueira Leao <ericcleao@gmail.com>

ericcleao committed:
feature/12875161 Update gradle version

- Version code updated from 20 to 21
- Version name updated from 2.5.2 to 2.6.0

Signed-off-by: Eric Cerqueira Leao <ericcleao@gmail.com>

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/camera/CameraFragment.kt:
```diff
@@ -0,0 +1,513 @@
+        super.onViewCreated(view, savedInstanceState)
+
+        mSaveOnlySnackbar = Snackbar.make(mCameraFragmentBinding.root,
+                getString(R.string.camera_snackbar_save_only_mode_message),
```
change this strings to be dinamically, texts should be provided by those who use the library

ericcleao replied:
Fixed

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/camera/CameraFragment.kt:
```diff
@@ -0,0 +1,513 @@
+                getString(R.string.camera_snackbar_save_only_mode_message),
+                Snackbar.LENGTH_SHORT)
+        mStantSaveModeSnackbar = Snackbar.make(mCameraFragmentBinding.root,
+                getString(R.string.camera_snackbar_stant_save_mode_message),
```
change this strings to be dinamically, texts should be provided by those who use the library

ericcleao replied:
Fixed

denisvieiradev commented on cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/camera/CameraFragment.kt:
```diff
@@ -0,0 +1,513 @@
+
+        changeSavePicturesReason.setOnClickListener{
+            mSaveMode = if (mSaveMode.mode == STANT_MODE) {
+                changeSavePicturesReason.setImageDrawable(ContextCompat.getDrawable(context!!, R.drawable.ic_stant_disabled))
```
this icons can be provided by those use the library, add dinamical variable to icon

ericcleao replied:
Fixed

ericcleao committed:
feature/12875161 Update camera fragment and card show taken picture view

- Add "enable save only mode" method at card show taken picture to make this new feature optional and editable
- When the app want to add this mode to the camera it should call this method and send the icons and messages that will be shown

ericcleao committed:
feature/12875161 Remove stant icon and snackbar message from the lib

-

Signed-off-by: Eric Cerqueira Leao <ericcleao@gmail.com>
